src/SharpLearning.InputOutput/Csv/CsvRowExtensions.cs

```diff
@@ -12,6 +12,73 @@ namespace SharpLearning.InputOutput.Csv
     /// </summary>
     public static class CsvRowExtensions
     {
+        /// <summary>
+        /// Combines two IEnumerables based on column header names. Matching rows are combined and parsed on. 
+        /// </summary>
+        /// <param name="thisRows"></param>
+        /// <param name="otherRows"></param>
+        /// <param name="key1"></param>
+        /// <param name="key2"></param>
+        /// <param name="removeRepeatedColumns">Should repeated columns be removed</param>
+        /// <returns></returns>
+        public static IEnumerable<CsvRow> KeyCombine(this IEnumerable<CsvRow> thisRows, IEnumerable<CsvRow> otherRows, string key1, string key2, bool removeRepeatedColumns = true)
+        {
+            var newColumnNameToIndex = thisRows.First().ColumnNameToIndex.ToDictionary(k => k.Key, k => k.Value);
```

Comment:
The new method overload of KeyCombine shares the generation of the `newColumnNameToIndex` and `columnIndicesToKeep` with the original implementation. To avoid code duplication, please refactor that part to the code into a new private static method, that the two methods can share. something like:

`static void CreateNewColumnNameToIndex(IEnumerable<CsvRow> thisRows, IEnumerable<CsvRow> otherRows, 
 bool removeRepeatedColumns, out Dictionary<string, int> newColumnNameToIndex, out int[] columnIndicesToKeep)
`

Since the method creates both `newColumnNameToIndex` and `columnIndicesToRemove`, these have to be out paramters. You are more than welcome to find a better name for the method :-)